src/QsFmt/Formatter/SyntaxTree/Statement.fsi

```diff
@@ -255,36 +275,160 @@ and internal For =
         Block: Statement Block
     }
 
+/// <summary>
+/// The concluding section of an <c>until</c> statement.
+/// </summary>
+and internal UntilCoda =
+
+    /// The semicolon.
+    | Semicolon of Terminal
+    
+    /// <summary>
+    /// The <c>fixup</c> of an <c>until</c> statement.
+    /// </summary>
+    | Fixup of BlockStatement
+
+/// <summary>
+/// An <c>until</c> statement.
+/// </summary>
+and internal  Until =
+    {
+        /// <summary>
+        /// The <c>until</c> keyword.
+        /// </summary>
+        UntilKeyword: Terminal
+        
+        /// <summary>
+        /// The condition under which to exit the preceding <c>repeat</c> block.
+        /// </summary>
+        Condition: Expression
+        
+        /// <summary>
+        /// The concluding section, possibly containing a <c>fixup</c> block.
+        /// </summary>
+        Coda: UntilCoda
+    }
+
+/// The concluding section of a qubit declaration.
+and internal QubitDeclarationCoda =
+
+    /// The semicolon.
+    | Semicolon of Terminal
+
+    /// The block of statements after the declaration.
+    | Block of Statement Block
+
+/// A qubit declaration statement.
+and internal QubitDeclaration =
+    {
+        /// The kind of qubit declaration.
+        Kind: QubitDeclarationKind
+
+        /// The keyword used in the declaration.
+        Keyword: Terminal
+
+        /// Optional open parentheses.
+        OpenParen: Terminal option
+
+        /// The qubit binding.
+        Binding: QubitBinding
+
+        /// Optional close parentheses.
+        CloseParen: Terminal option
+
+        /// The concluding section.
+        Coda: QubitDeclarationCoda
+    }
+
 /// A statement.
 and internal Statement =
+
+    /// An expression statement.
+    | ExpressionStatement of ExpressionStatement
+
+    /// <summary>
+    /// A <c>return</c> statement.
+    /// </summary>
+    | Return of SimpleStatement
+
+    /// <summary>
+    /// A <c>fail</c> statement.
+    /// </summary>
+    | Fail of SimpleStatement
+
     /// <summary>
     /// A <c>let</c> statement.
     /// </summary>
-    | Let of Let
+    | Let of BindingStatement
 
     /// <summary>
-    /// A <c>return</c> statement.
+    /// A <c>mutable</c> declaration statement.
     /// </summary>
-    | Return of Return
+    | Mutable of BindingStatement
 
-    /// A qubit declaration statement.
-    | QubitDeclaration of QubitDeclaration
+    /// <summary>
+    /// A <c>set</c> statement.
+    /// </summary>
+    | SetStatement of BindingStatement
```

Comment:
Set and update statement seem to be the only thing called statement; aligning closer with Let and Mutable, it would make sense to call them Set, Update, and UpdateWith.

Reply:
Added Statement to the end of all the statement data structures.